internal/workflow/preprocessing.go

```diff
@@ -110,6 +111,29 @@ func (w *PreprocessingWorkflow) Execute(
 
 	localPath := filepath.Join(w.sharedPath, filepath.Clean(params.RelativePath))
 
+	// Unbag the SIP if it's a BagIt bag.
+	if fsutil.FileExists(filepath.Join(localPath, "bagit.txt")) {
```

Comment:
It might be a good idea to wrap `fsutil.FileExists` with the SideEffect API or implement it as a local activity given that it relies on the local disk which is an unreliable resource. For the reader unclear about this problem:
* https://docs.temporal.io/workflows#intrinsic-non-deterministic-logic
* https://learn.temporal.io/tutorials/go/background-check/durable-execution/#intrinsic-non-deterministic-logic
* https://community.temporal.io/t/workflow-determinism/4027#intrinsic-4

workflowcheck doesn't seem able to identify this issue, but I think that's true in many other cases - they only seem to be paying attention to a subset of issues.
I haven't been actively following the work in this repo for a while, but I came across this pull request by chance while reviewing #97 and #98 yesterday - though I’m no longer looking at them, thanks!

Reply:
Hmm, good catch @sevein, I didn't have file I/O on my radar for non-deterministic code. :(  The Unbag activity itself is a no-op when the SIP is not a bag, but I'd rather not create and unbagging event if we aren't unbagging anything so I'll have to figure out how to conditionally create the event.